sky/resources.py

```diff
@@ -1379,9 +1388,6 @@ def __setstate__(self, state):
         if version < 2:
             self._region = None
 
-        if version < 3:
-            self._spot_recovery = None
```

Comment:
Is this necessary? If not, better to keep it in history?

Reply:
Just to make the history of field versions monotonic. It's more readable; otherwise, readers are left to guess what happened in this gap.

Reply:
nit: How about we comment out the version 3 check and add a comment here? Still, just to keep the code being executed is relevant. In the future, we may even want a policy that removes some of these backward compatibility checks as what we did for the other places (remove backward compatibility code after 2/3 major releases).

Reply:
I still think we should just keep the history. Purging old fields is a good idea, but we should do it in chunks, rather than leaving "holes".

For example, with it commented I notice that switching from this PR to the master shows
```
» sky spot launch exit 1 -y
...

Launching managed spot job 'sky-cmd' from spot controller...
Launching spot controller...
AttributeError: 'Resources' object has no attribute '_spot_recovery'
```